File: src/main/java/com/refinedmods/refinedstorage/api/storage/disk/IStorageDiskContainerContext.java
package com.refinedmods.refinedstorage.api.storage.disk;

import com.refinedmods.refinedstorage.api.storage.AccessType;

/**
 * Stores relevant information about the container where the {@link IStorageDisk} is in.
 */
public interface IStorageDiskContainerContext {
    /**
     * @return the access type
     */
    AccessType getAccessType();
}

File: src/main/kotlin/com/refinedmods/refinedstorage/api/storage/disk/IStorageDiskContainerContext.kt
package com.refinedmods.refinedstorage.api.storage.disk

import com.refinedmods.refinedstorage.api.storage.AccessType


/**
 * Stores relevant information about the container where the [IStorageDisk] is in.
 */
interface IStorageDiskContainerContext {
    /**
     * @return the access type
     */
    fun getAccessType(): AccessType?
}